### ISSUE TITLE
Profiler context menu is missing or the project context menu itself is infinitely repeated

### ISSUE THREAD

**Issue by tyoma** [tyoma/micro-profiler#25]
Installed for VS2015.  When I right-click on project 64 bit Release in Solution Explorer no menu appears, where before the installation it did appear.

When I right-click on 32 bit Release project I get the default menu but with the standard entries infinitely duplicated in a continuous, never-ending context menu which, in any case, doesn't include the 'Enable Profiling' option that should have been added by the micro-profiler installer, which appeared to have run correctly.

Any ideas?

**Comment by pboulton** [comment]
Please see attached zip which hopefully contains the files you've requested.

I have noticed the following in my further testing:

(1) The 'infinite' menu only appears when devenv.exe is launched as Administrator.  It appears correctly when launched as a standard, non-Administrator user.
(2) In both cases, the popup menu does not appear for my Release (Unicode) + x64 build configuration.  Note that I mean the whole menu, not just the 'Enable Profiling' menu option.  I have a feeling this may be due to having previously run the built in VS2015 profiler.
(3) So the only time things work as expected is launching devenv.exe as non-Administrator and for Release + Win32 build configuration (which has not been previously profiled using the built in VS2015 profiler).

I hope that's all clear!  Please let me know if you need any further information.  Thanks!

[MicroProfiler.zip](https://github.com/tyoma/micro-profiler/files/569286/MicroProfiler.zip)

**Comment by tyoma** [comment]
Thanks Peter! The logs don't show much about micro-profiler, but I spotted some errors  with other packages. Indeed, the screenshot you've attached and the details about admin/non-admin usage seem to be more helpful, I'll try to conduct testing in that way...

Btw, did you try it with other VS versions before?

**Comment by pboulton** [comment]
Hi Artem,

I have VS2010 and VS2015 (and VC6!) installed.  The project I'm testing with MicroProfiler is a VS2010 project which I'm using in VS2015 with the 'General' --> 'Platform Toolset' setting set to  VS2010.

I haven't tested MicroProfiler in VS2010 and, on this project, I'm reluctant to go back as it may cause problems re-using it on VS2015.  It's the project that earns the money that feeds my family!  :-).

To the best of my knowledge, the only non-core things I have in VS2015 are VisualAssist, latest build and MicroProfiler (though I have uninstalled it while we, hopefully, find a solution).

Many thanks for your help on this.  I really do appreciate it but I wouldn't want you spending too much time on it unless it's something that helps you improve MicroProfiler for everyone!

All the best,

Pete

**Comment by tyoma** [comment]
Got it. I'll see how it integrates with VA. Btw I wasn't insisting on using earlier VS, just curious, if MP's integration worked normally in that enviroment...
Will MP be helpful to you w/o integration (i.e. with manual configuration I outlined)?

**Comment by pboulton** [comment]
Hi tyoma.  

I'm afraid I just don't want to go back to VS2010 as VS2015 is much nicer!  I'm not that keen on manual integration either, to be honest.  I'd rather keep my source code 'clean' - sorry.  :-(  [Unless, in using the MicroProfiler, I find that limiting the profiling to a small number of source files within the project is necessary.]

Do you think the issue could be related to my specific project?  I have uninstalled MicroProfiler but perhaps I should reinstall and generate a new project, to see if I get the same problems.  Would that help?

All the best,

Pete

**Comment by tyoma** [comment]
haven't got a chance to check it yet, but my best bet is that it is something to do with USERness OR with vspackage registration sequence... You see, the actual registration of a VSPackage (any of them) occurs when you first run devenv after it was run with /setup switch (if I remember it correctly). And it occurs on a per-user basis. so my guess is that after the first run (when you're actually able to see integration), not all VSPackages complete their installation sequences. As a result, on the next run the installation continues, which leads to duplication of menus. Maybe, it is the problem of micro-profiler.dll itself. I'm going to investigate this.

**Comment by pboulton** [comment]
IIRC I had to run the devenv.exe /setup from an Administrator command prompt because it failed when I ran it in standard mode.  I'll try again in the morning!

**Comment by tyoma** [comment]
'/setup' requires that

**Comment by pboulton** [comment]
This morning I reinstalled MicroProfiler and created a new MFC project in VS2015.  Everything worked as documented with MicroProfiler.  Therefore it would seem that the issue I've experienced is somehow related to my original project, which (IIRC) has migrated over the years from VC6 through VS2008, VS2010 to VS2015.

I'm happy to send you whatever solution files you ask for, just not my cpp/h source files!

Equally, given the issue appears to be related to my specific project I would totally understand if you prefer to draw a line here as I clearly isn't a major MicroProfiler issue and, indeed, may not be a MicroProfiler issue at all.

Many thanks again for the diligence you've applied to helping me - it really is most appreciated.

All the best,

Pete
